dist-git-to-src-git/updates.md

```diff
@@ -0,0 +1,126 @@
+      - diff of patches / revert+new change
+    - save `{"patch": "name.patch", index=42}` with the commit
+    - line with patch should be already commented
+
+### Reverse
+
+- When going back from commit to patch:
+  - In case of commit without metadata:
+    - Generate a patch file and add it to the spec-file.
+  - In case of `index >= 0` in the metadata:
+    - Check the existence of the `name.patch`:
+      - If not present:
+        - Uncomment the `index` line in the spec-file.
+        - Generate `name.patch` from the commit.
+      - If present:
+        - Generate a new path from the commit and merge it into `name.patch`.
+  - In case of `index == -42`:
+    - Remove the `name.patch`.
+    - Remove the line `42` in the spec-file.
+
+
+### Saving metadata
+
+We have various way, how to save metadata for regeneration:
+
+- Enhance packit config file:
+  - ➕ editable by users
+  - ➕ can be seen/reviewed
+  - ➖ is not connected with the commit
+- git note (see [Git Notes research](../git_notes) for more info):
+  - ➕ attached to the git commit
+  - ➕ can be changed, but not so easily as a regular file
+  - ➖ not pulled automatically
+- git-message
+  - We can have strict format of the git-message (e.g. yaml).
+  - ➕ attached to the git commit
+  - ➕ pulled automatically
+  - ➖ if we need to extend the info, we can loose readability
+  - ➖/⁉ cannot be edited when pushed
+- other file
+  - ➕ editable by users
+  - ➕ can be seen/reviewed
+  - ➖ is not connected with the commit
+  - ➖ another file
+
+### Conclusion of proposal 1
+
+- We need to add metadata to the source-git commits.
+  - Can be used for other information.
+  - Depending on the saving mechanism (above) it can reduce/break the usual git usage.
+- ➕ The git history is clear.
+- ➖ A lot of work needs to be done.
+  - Multiple scenarios; lot of corner cases.
+- ➖ It will be hard to correct/improve the workflow.
+- ➕ Reverted dist-git still preserves the conditions, naming and location of the patches.
+- ➖ Generated source-git will contain all the patches.
+  - Various improvements and strategies possible and doable later
+    (e.g. avoid both of if-else branches, expand macros before getting of the patches).
+
+
+## Proposal 2
+
+- Easy solution that mimics the history overwriting without force push.
+- Regenerate the source-git from scratch and use
+  [ours](https://git-scm.com/docs/merge-strategieshttps://git-scm.com/docs/merge-strategies)
+  merging strategy to merge the new version on top of the old version ignoring its content.
+  - Alternatively, we can `revert` all commits and apply the new ones to have linear history.
+    - All those newly-created commits can be squashed together to have only one new commit.
+    - Will the revert of this work?
+- We need to make sure that we go by the correct branch when reverting the content to the dist-git style.
+  - Saving metadata (like above) can help us with that.
+
+### Conclusion of proposal 2
+
+- ➖ Git tree is unreadable.
```

Comment:
this is a huge downside, since one of the main reasons we are doing source-git is to finally use git as it's meant to use

Reply:
I've figured out some improvements. I'll add some pictures.